docs/serialisation/protobuf.md

```diff
@@ -1,86 +1,172 @@
+
+Additionally, in the context of `alloy`, the presence of the `@protoWrapped` trait is interpreted as requiring the primitive to be wrapped in a one-field message.
+
+For instance :
+
+```smithy
+@protoWrapped
+string MyString
+```
+
+would be converted to
+
+```proto
+message MyString {
+  string value = 1;
+}
+```
+
+When converting .smithy IDL files to .proto IDL, types from the `google.protobuf` library and the `alloy.protobuf` library can be used.
+Using `protoWrapped` is interesting as it permits the distinction between the absence of a value and the presence of a default value.
+
+| Smithy type          | @protoNumType |                                 |
+| -------------------- | ------------- | ------------------------------- |
+| float                | N/A           | google.protobuf.FloatValue      |
+| blob                 | N/A           | google.protobuf.BytesValue      |
+| boolean              | N/A           | google.protobuf.BoolValue       |
+| double               | N/A           | google.protobuf.DoubleValue     |
+| bigDecimal           | N/A           | alloy.protobuf.BigDecimalValue  |
+| bigInteger           | N/A           | alloy.protobuf.BigIntegerValue  |
+| string               | N/A           | google.protobuf.StringValue     |
+| integer, byte, short | N/A           | google.protobuf.Int32Value      |
+| integer, byte, short | FIXED         | alloy.protobuf.FixedInt32Value  |
+| integer, byte, short | FIXED_SIGNED  | alloy.protobuf.SFixedInt32Value |
+| integer, byte, short | SIGNED        | alloy.protobuf.SInt32Value      |
+| integer, byte, short | UNSIGNED      | google.protobuf.UInt32Value     |
+| long                 | N/A           | google.protobuf.SInt64Value     |
+| long                 | FIXED         | alloy.protobuf.Fixed64Value     |
+| long                 | FIXED_SIGNED  | alloy.protobuf.SFixed64Value    |
+| long                 | SIGNED        | alloy.protobuf.SInt64Value      |
+| long                 | UNSIGNED      | google.protobuf.UInt64Value     |
+
+#### alloy.proto#protoNumType
+
+Integer and Long shapes can be annotated with the `@alloy.proto#protoNumType` in order to signal what encoding should be used during protobuf serialisation.
+
+- SIGNED
+- UNSIGNED
+- FIXED
+- FIXED_SIGNED
+
+See [here](https://protobuf.dev/programming-guides/proto3/#scalar) for documentation about these encodings.
+
 
 #### UUIDs
 
-`alloy` gives special meaning to the `alloy#UUID` shape in the context of protobuf : `UUIDs` are supposed to be encoding by means of a message containing two long values, which is more optimal than serialising a string.
+By default, string shapes annotated with `@uuidFormat` are serialised as protobuf strings. However, a `alloy.proto#protoCompactUUID` trait is provided, which signals that the serialised form should be a message containing two int64 values :
 
 Smithy:
 
 ```smithy
+use alloy#uuidFormat
+use alloy.proto#protoCompactUUID
+
+@protoCompactUUID
```

Comment:
smithy translate did that for shapes with the type:`alloy#UUID`

https://github.com/disneystreaming/smithy-translate/blob/main/modules/proto-examples/protobuf/demo/definitions.proto#L24

Reply:
I'm aware. We shouldn't use shapeIds to discriminate behaviour, it's pretty unsound. Using traits is much better. 

I guess your point is that we could apply that trait to the `alloy#UUID` idea to retain the current behaviour.

Reply:
I did not know you were aware, I agree with you that the trait is a better approach now that I know better!
But as jeff points out, I don't think introducing a dependency between the two artifacts is desired so I would just leave it like that